java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -272,6 +272,12 @@ private void prune(Block inactiveBlock) {
         }
         falseBlock = inactiveBlock.successors.iterator().next();
       }
+      if (inactiveBlock.equals(successorWithoutJump)) {
+        if (inactiveBlock.successors.size() != 1) {
```

Comment:
What about using rather a `com.google.common.base.Preconditions` rather than a `if` with a `throw`?:
```java
Preconditions.checkState(
  inactiveBlock.successors.size() == 1, 
  "SuccessorWithoutJump successor must be replaced by a unique successor!");
```

I would also prefer to have this in a distinct commit. You are changing the behavior of the CFG, and it's going to be lost in a "fix unreachable code" commit